spec/liquid_tags/unified_embed_tag_spec.rb

```diff
@@ -0,0 +1,17 @@
+require "rails_helper"
+
+RSpec.describe UnifiedEmbedTag, type: :liquid_tag do
+  it "delegates parsing to the link-matching class" do
+    link = "https://gist.github.com/jeremyf/662585f5c4d22184a6ae133a71bf891a"
+
+    allow(GistTag).to receive(:new).and_call_original
+    parsed_tag = Liquid::Template.parse("{% embed #{link} %}")
+    expect { parsed_tag.render }.not_to raise_error
```

Comment:
The expectation that `GistTag` was called seems to be missing from this spec. Adding a suggestion in case this is the expected input (only change is to add the last line):

```suggestion
    allow(GistTag).to receive(:new).and_call_original
    parsed_tag = Liquid::Template.parse("{% embed #{link} %}")
    expect { parsed_tag.render }.not_to raise_error
    expect(GistTag).to have_received(:new).with(link)
```